testsuite/integration/basic/src/test/java/org/jboss/as/test/integration/jpa/webtxem/TestServlet.java

```diff
@@ -48,13 +48,28 @@ protected void doGet(HttpServletRequest req, HttpServletResponse resp) throws Se
 
                 if (mode.equals("write")) {
                     WebJPAEntity f = new WebJPAEntity();
-                    f.setId(1L);
+                    // Pasting from https://docs.jboss.org/hibernate/orm/6.6/migration-guide/migration-guide.html:
+                    // "
+                    // Merge versioned entity when row is deleted
+                    // Previously, merging a detached entity resulted in a SQL insert whenever there was no matching row in the database (for example, if the object had been deleted in another transaction). This behavior was unexpected and violated the rules of optimistic locking.
+                    //
+                    // An OptimisticLockException is now thrown when it is possible to determine that an entity is definitely detached, but there is no matching row. For this determination to be possible, the entity must have either:
+                    //
+                    // *  a generated @Id field, or
+                    // *  a non-primitive @Version field.
+                    //
+                    // For entities which have neither, it’s impossible to distinguish a new instance from a deleted detached instance, and there is no change from the previous behavior.
+                    // "
+                    // Previously, we specified the WebJPAEntity id value but as mentioned ^ we can no longer do that.
+                    // f.setId(1L);
```

Comment:
@marko-bekhta I want to make sure I understand this correctly.

First, calling setXXX for a property annotated with @ GeneratedValue is an anti-pattern. So what we've been doing here is pretty broken.

With ORM 6.6, because the property has @ GeneratedValue, ORM can definitively determine the entity is detached is no DB record with key '1L', so this fails with an OLE.

Right? 

I think the key point is 'what we've been doing here is pretty broken'. This kind of app needs migration but such an app was always flawed.

Reply:
Nit, only if you happen to change something for some other reason, and even then it's optional:

Perhaps:

"Previously, we specified the WebJPAEntity id value but as mentioned ^ because the @Id field is annotated with @GeneratedValue we can no longer do that."

I had to parse things a bit to understand this; maybe that would have helped.

No big deal.

Reply:
> I think the key point is 'what we've been doing here is pretty broken'. This kind of app needs migration but such an app was always flawed.

yes 👍🏻 
I also opened this follow-up PR https://github.com/wildfly/wildfly/pull/18207 to update the comment (including your suggestion) and keep this one ready for merging without waiting for another CI run.